lib/kennel/models/monitor.rb

```diff
@@ -108,6 +108,16 @@ def as_json
         @as_json = data
       end
 
+      def resolve_linked_tracking_ids(id_map)
+        if as_json[:type] == "composite"
+          begin
+            as_json[:query].gsub!(/%\{(.*?)\}/) { id_map.fetch($1) }
+          rescue KeyError => e
+            Kennel.err.puts("Unable to find #{e.key} in existing monitors (they need to be created first to link them)")
```

Comment:
reuse `resolve_link` (by moving it into base) and don't use fetch, but do `gsub { resolve_link($1) || $1` ... make sure that query is still valid when missing so the initial creation does not get a 400 error

Reply:
Not actually sure what a sensible valid default query would be for a composite monitor, since it seems to require a valid monitor id.